src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -443,6 +458,56 @@ def _upload_assembly_files_in_parallel(
     return failed_names
 
 
+def _gen(
+        conf: Conf,
+        wait_to_upload_assemblies: dict[str, str]
+) -> Generator[tuple[tuple[str, ...], tuple[str, ...], tuple[str, ...], int], None, None]:
+    """
+    Generator function to yield the assembly files to upload.
+    """
+    assembly_files_len = len(wait_to_upload_assemblies)
+    assembly_names = tuple(wait_to_upload_assemblies.keys())
+    host_assembly_dirs = tuple(wait_to_upload_assemblies.values())
+    container_internal_assembly_paths = tuple(
+        [
+            os.path.join(JOB_DIR_IN_ASSEMBLYUTIL_CONTAINER, assembly_name)
+            for assembly_name in assembly_names
+        ]
+    )
+
+    start_idx = 0
+    cumsize = 0
+    # to avoid memory issue, upload 1GB * MEMORY_UTILIZATION of assembly files at a time
+    max_cumsize = MAX_DATA_SIZE * MEMORY_UTILIZATION
```

Comment:
I'm not sure I understand the memory utilization factor here. Surely that would be something like

```
max_cumsize = min(MAX_DATA_SIZE, getMemorySize() * MEMORY_UTILIZATION)
```

And 1% of memory seems pretty low

Reply:
max_cumsize is pre-calculated here. MEMORY_UTILIZATION a basically a percentage ideally closer to 100%. I set  it to 1% because I benchmarked only on 50 assembly files and wanted each worker had something to work on.

Reply:
Addressed in AssemblyUtil

Reply:
:+1: